search/lib/stac/extensions/fields.js

```diff
@@ -1,14 +1,9 @@
 
 const _ = require('lodash');
 
-function convertStacFieldsQueryToObject (fieldsQuery) {
```

Comment:
Let me know if you think removing this public method (and making apply the only public method) is a mistake.

Reply:
While it doesn't crash @dylnclrk , I think this actually breaks the `getSearch()` method.  I'm digging further to see how this may have regressed.

Reply:
Ok @dylnclrk , sure enough this has caused a regression with the fields extension.  In the new private method, it is never getting called with the GET request.  Please try using this request and see if you can figure out where the regression is taking place 👍

Reply:
```
## GET Provider Search for Collection - /:providerId/search?collections=:collectionId
# Run a search against CMR using the STAC API endpoint interface.
curl "http://localhost:3000/cmr-stac/GES_DISC/search?collections=C1237113465-GES_DISC&fields=properties.datetime,-assets"
```

Reply:
^^ Should OMIT the assets field.

Reply:
This is looking great now!  Thank you for the fix.